### PR TITLE
Fix submodule parsing

### LINKED ISSUE #32568: Submodules not linked if file format not in default order
### Description

A submodule that is listed in .gitmodules as,

```
[submodule "jakarta-servlet"]
        url = ../../ALP-pool/jakarta-servlet
        path = jakarta-servlet
```

is not linked in the repository view. A submodule that is listed in the other order is linked correctly.

```
[submodule "jakarta-servlet"]
        path = jakarta-servlet
        url = ../../ALP-pool/jakarta-servlet
```

The gitmodules.txt doesn't seem to impose any particular order on values.

NOTE: Demo site is down at this second so can't check if I can reproduce it there.

### Gitea Version

1.22.2

### Can you reproduce the bug on the Gitea demo site?

Yes

### Log Gist

_No response_

### Screenshots

image

### Git Version

_No response_

### Operating System

_No response_

### How are you running Gitea?

It's built and running with systemd, using core git tools

### Database

PostgreSQL

### PR DESCRIPTION
**Pull request by lunny**
Fix #32568, parse `.gitmodules` correctly

### CODE REVIEW

**Review by delvh — APPROVED** [review]

**Review by wxiaoguang — DISMISSED** [review]
Wrong.

Gitea never uses "go-git" in its non-gogit version.

**Comment by lunny** [comment]
> Wrong.
> 
> Gitea never uses "go-git" in its non-gogit version.

Any better idea? Could you recommend another library or should we parse it ourselves?

**Comment by wxiaoguang** [comment]
It is quite simple, just write one?

File: modules/git/commit.go [hunk 1]
```diff
@@ -375,32 +374,13 @@ func (c *Commit) GetSubModules() (*ObjectCache, error) {
...
-				c.submoduleCache.Set(path, &SubModule{path, strings.TrimSpace(fields[1])})
-				ismodule = false
-			}
-		}
-	}
-	if err = scanner.Err(); err != nil {
-		return nil, fmt.Errorf("GetSubModules scan: %w", err)
-	}
 
+	r := io.LimitReader(rd, 100*1024) // limit to 100KB
```
Comment (AdamMajer) [inline modules/git/commit.go]: We use submodules in funny way (to represent packages in a Linux distribution :smile: ) and our file is 250kB (5k submodules). 1MB for openSUSE Tumbleweed is possible here. Larger if not using relative URLs

Is there really a need to impose such small limit like this here? Previously there wasn't a limit.

Comment (AdamMajer) [reply modules/git/commit.go]: This should read "2MB is possible for openSUSE Tumbleweed" :laughing:

Comment (wxiaoguang) [reply modules/git/commit.go]: let's remove the limit and keep the old logic.

**Review by wxiaoguang — APPROVED** [review]

File: modules/git/commit_info.go [hunk 2]
```diff
@@ -7,5 +7,5 @@ package git
 type CommitInfo struct {
 	Entry         *TreeEntry
 	Commit        *Commit
```
Comment (lunny) [inline modules/git/commit_info.go:9]: Looks like `*Commit` could be removed.

Comment (wxiaoguang) [reply modules/git/commit_info.go:9]: The `Commit` was removed from `CommitSubModuleFile` because it was never used.

**Pull request merged**